frontend/views/components/Modal/ModalTemplate.vue

```diff
@@ -49,27 +50,240 @@ export default {
     this.isActive = true
     sbp('okTurtles.events/on', OPEN_MODAL, this.openModal)
     sbp('okTurtles.events/on', CLOSE_MODAL, this.closeModal)
+    window.addEventListener('keyup', (e) => {
+      if (e.key === 'Escape') {
+        this.close()
+      }
+    })
   },
   beforeDestroy () {
+    window.removeEventListener('keyup', null)
     sbp('okTurtles.events/off', OPEN_MODAL, this.openModal)
     sbp('okTurtles.events/off', CLOSE_MODAL, this.closeModal)
   },
   methods: {
     close (e) {
-      // To remove when @keyup.esc Vue bug is fixed
-      const isEscapeKey = e.key === 'Escape'
-      // Fix the bug where keyboard event trigger click type event
-      const isCloseButtonClick = e.type === 'click' && e.screenX !== 0
-      if (isEscapeKey || isCloseButtonClick) {
-        sbp('okTurtles.events/emit', 'close-modal')
-      }
+      sbp('okTurtles.events/emit', 'close-modal')
     },
     openModal () {
       this.isActive = true
     },
     closeModal () {
       this.isActive = false
+      if (location.href.indexOf('?modal') > 0) {
+        history.pushState(null, null, location.href.split('?modal=')[0])
```

Comment:
This is not the best or simplest or bug-free way to do this. The issue here is that by splitting the string at `?modal=`, and then passing in the first element of the array, the code fails to:

- Account for other URL variables that might come after it
- It assumes that `modal` will always be the first part of the URL, but the code on line 32 of Modal.vue simply appends `?modal=` to the URL, which could result in gibberish invalid URLs when the URL already has arguments appended  to it

So this approach is not workable. Either you must use (or create) a valid URL manipulating library, or a non-URL approach must be taken.

Another issue is the use of `window.localStorage` to keep track of the state of the popup. That is code smell. We have two kinds of state management in the app, and they are both described in our [Style Guide Data Model Rules](https://github.com/okTurtles/group-income-simple/blob/master/docs/Style-Guide.md#group-income-data-model-rules). Adding a third type of state management haphazardly here is again, code smell.

I recommend ignoring the back button for now. We might not use it anyway. And if we do, we will have to design and implement a coherent system for it, which will be its own issue and its own PR.

Reply:
Good comments Greg!
You raised a fair point about the ?modal=.
I did that intentionally however because I always want this to be the first argument, no matter what. I don't see any reason why other arguments should be coming before, and if there is it will raise that there is a problem in the URL structure. My recommendation is that we keep it this way :) Otherwise the simple solution to this problem is to remove the '?'. I don't see atm what a 'manipulating library' would do. Maybe you can elaborate on that point?

Url approach is required by @mmbotelho in the ticket for a valid point. I'm personally very annoyed when devs don't let me use the back option on mobile phone to remove the popup (which appear as a new page on mobile).
Also, we might want to redirect people to page with modals: for example if we link to a particular section of the settings in the user documentation.

An alternative is to NOT use modal, which is from my point of view the best approach, but that mean changing quite a bite the UX and design that @mmbotelho did.

Concerning the localStorage I agree. I should use the store. That was an old programmer reflex  when dealing with browser history... ;)

I don't think we need another ticket for this has this is a very quick change.

Reply:
Modals should not have URLs unless they are absolutely needed. Remember, this is an app, not a normal website. Modals do not have URLs in Slack, Figma, Discord, or any other app that I'm aware of, and they probably won't have URLs in Group Income either.

However, *if* we do add URLs for modals, then we will do it very carefully, and so an issue will need to be opened for that, and a clear problem description written up.

In general, using the URL to pass parameters is something that will require a separate issue and separate PR for, because it has to be well thought out. Simply manipulating the URL randomly with string concatenation in random places in the code is a definite no-go, because it will lead to ugly code, and lots and lots of bugs.

Reply:
>  I don't see atm what a 'manipulating library' would do. Maybe you can elaborate on that point?

A URL manipulating library makes it possible to manipulate the URL (exactly what you're trying to do), but safely, and without directly touching the URL and manipulating it yourself. It ensures the URL is well-formed, and makes it easy to add/remove/get parameters and their values.

However, again, we should not dive into using such a thing without a *lot* of thought and discussion first.

Reply:
Ok! That was only my recommendations... I'll remove my solution and I'll let you create a ticket for that with all the requirements.

Reply:
Yet another critical reason why a lot of thought must go into this (and a separate issue created, if they are deemed necessary), is that manipulating the URL and browser history manually as  this PR does **will** introduce severe conflicts with any other existing URL and history manipulating libraries we use.

We already use one such library, and this PR would be in conflict with it: [`vue-router`](https://router.vuejs.org)

VueRouter has its own way of passing parameters through the URL and interacting with them. We would need to heavily think about and weigh the pros & cons of different approaches, i.e. whether this use case would need URL parameters at all. If it was deemed necessary, then we would need to decide how we would use them, i.e. via VueRouter's API, or via a different library. If via a different library, we would have to research how that library should interact with VueRouter and VueX, etc.

Reply:
>  I'll remove my solution and I'll let you create a ticket for that with all the requirements.

👍 

Re the ticket: I'm not convinced that the Modal needs to be tied to the URLs in the first place, so I wouldn't know what kind of ticket to create. So if you have a Problem/Solution statement, or if @mmbotelho does, then one of you should create that ticket and we can discuss it

Reply:
I've fixed all the history manipulation using basic vue router watch and query parameter.
Great work on reviewing the code Greg!